src/hammer-vlsi/hammer_vlsi/hammer_submit_command.py

```diff
@@ -0,0 +1,188 @@
+            database.set_setting(tool_namespace + ".submit_command_settings", None)
+
+        name = database.get_setting(tool_namespace + ".submit_command", nullvalue="none")
+
+        if name == "none" or name == "bare":
+            if database.get_setting(tool_namespace + ".submit_command_settings", nullvalue="") != "":
+                raise ValueError("Unexpected " + tool_namespace + ".submit_command_settings for HammerBareSubmitCommand. Did you forget to set " + tool_namespace + ".submit_command?")
+            return HammerBareSubmitCommand()
+        elif name == "lsf":
+            submit_command = HammerLSFSubmitCommand()
+            submit_command.read_settings(tool_namespace, database)
+            return submit_command
+        else:
+            raise NotImplementedError("Submit command key for " + tool_namespace + ": " + name + " is not implemented")
+
+
+class HammerBareSubmitCommand(HammerSubmitCommand):
+
+    def submit(self, args: List[str], env: Dict[str,str], logger: HammerVLSILoggingContext, cwd: str = None) -> str:
+        """
+        Submit the job to the job submission system. This function MUST block until the command is complete.
+        This implementation just runs the command on the host.
+
+        :param args: Command-line to run; each item in the list is one token. The first token should be the command to run.
+        :param env: The environment variables to set for the command
+        :param logger: The logging context
+        :param cwd: Working directory (leave as None to use the current working directory).
+        :return: The command output
+        """
+        # Short version for easier display in the log.
+        PROG_NAME_LEN = 14 # Capture last 14 characters of the command name
+        if len(args[0]) <= PROG_NAME_LEN:
+            prog_name = args[0]
+        else:
+            prog_name = "..." + args[0][len(args[0])-PROG_NAME_LEN:]
+        remaining_args = " ".join(args[1:])
+        if len(remaining_args) <= 15:
+            prog_args = remaining_args
+        else:
+            prog_args = remaining_args[0:15] + "..."
+        prog_tag = prog_name + " " + prog_args
+
+        output_buf = ""
+
+        logger.debug("Executing subprocess: " + ' '.join(args))
+        subprocess_logger = logger.context("Exec " + prog_tag)
+        proc = subprocess.Popen(args, shell=False, stderr=subprocess.STDOUT, stdout=subprocess.PIPE, env=env, cwd=cwd)
+        atexit.register(proc.kill)
+        # Log output and also capture output at the same time.
+        while True:
+            line = proc.stdout.readline().decode("utf-8")
+            if line != '':
+                subprocess_logger.debug(line.rstrip())
+                output_buf += line
+            else:
+                break
+        # TODO: check errors
+
+        return output_buf
+
+    def read_settings(self, tool_namespace: str, database: HammerDatabase) -> None:
+        assert("Should never get here")
```

Comment:
I would prefer something like "Bare submission command does not have settings"

Reply:
OK, added.